msal/src/Microsoft.Identity.Client/AuthenticationResult.cs

```diff
@@ -58,8 +58,16 @@ internal AuthenticationResult(MsalAccessTokenCacheItem msalAccessTokenCacheItem,
             _msalIdTokenCacheItem = msalIdTokenCacheItem;
             if (_msalAccessTokenCacheItem.HomeAccountId != null)
             {
-                Account = new Account(AccountId.FromClientInfo(_msalAccessTokenCacheItem.ClientInfo),
-                    _msalIdTokenCacheItem?.IdToken?.PreferredUsername, _msalAccessTokenCacheItem.Environment);
+                if (_msalAccessTokenCacheItem.TenantId == null)
```

Comment:
> if (_msalAccessTokenCacheItem.TenantId == null) [](start = 15, length = 48)

not sure how this will work in non ADFS cases. @somska @bgavrilMS, thoughts on this?

Reply:
@somska can confirm, but AFAIK clients migrated from ADAL v3 will have ClientInfo null => tenantId is null. 
@reed1995 - will ADFS always have an UPN there? Maybe we could put UPN in the condition?

---
In reply to: [227041247](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1293#discussion_r227041247) [](ancestors = 227041247)

Reply:
Added an IsAdfs property on the token cache item and use that in the conditional.